lib/compilation_engine.rb

```diff
@@ -257,10 +257,18 @@ def compile_expression
     compile_term
 
     while symbol_token?(*OP_SYMBOLS)
+      operator_symbol = @tokenizer.symbol
```

Comment:
This could work for `Seven` but not, I think, in general.

Imagine if the expression was `1 + 2 - 3 + 4` instead of `1 + (2 * 3)`; the resulting VM code should be something like:

```
push constant 1
push constant 2
add
push constant 3
sub
push constant 4
add
```

But you’re only emitting the VM code for the operator once at the end — even if the `while` loop has executed many times — and you’re only using the final value of `operator_symbol` so any earlier operators are ignored.

The solution to this problem isn’t obvious, but pages 231 & 232 explain the two-stage strategy: first build an AST representing the expression, then walk over that AST emitting values and operators in the correct order. Your shortcut happens to do the same thing for the special case of expressions containing exactly two atomic values separated by one operator, but not in other cases.